torch/utils/cpp_extension.py

```diff
@@ -740,6 +778,19 @@ def _check_abi(self):
                    'Please set `DISTUTILS_USE_SDK=1` and try again.')
             raise UserWarning(msg)
 
+    def _check_cuda_version(self):
+        if CUDA_HOME:
+            nvcc = os.path.join(CUDA_HOME, 'bin', 'nvcc')
+            cuda_version_str = subprocess.check_output([nvcc, '--version']).strip().decode()
+            cuda_version = re.search(r'release (\d+[.]\d+)', cuda_version_str)
+            if cuda_version is not None:
+                cuda_version = parse_version(cuda_version.group(1))
+                torch_cuda_version = parse_version(torch.version.cuda)
+                if cuda_version.major != torch_cuda_version.major:
+                    warnings.warn(CUDA_MISMATCH_MESSAGE.format(cuda_version, torch.version.cuda))
+        else:
+            warnings.warn(CUDA_NOT_FOUND_MESSAGE)
```

Comment:
same here? why issue a warning when it can tell the user what to do right away to fix the problem